subprojects/model-core/src/main/java/org/gradle/internal/instantiation/generator/AsmBackedClassGenerator.java

```diff
@@ -1137,6 +1142,13 @@ public void applyReadOnlyManagedStateToGetter(PropertyMetadata property, Method
                     applyRole();
                 }
 
+                String buildServiceName = getBuildServiceName(property);
```

Comment:
An attempt to resolve the property would fail due to lack of value or convention being set. Maybe we can wait and see if that is an actual source of confusion?

Reply:
Actually, we always set the convention, no matter the type of the property owner. But, except for a task's nested properties,  we do not add it to any task's required resources.